src/SignalR/clients/cpp/src/signalrclient/default_websocket_client.cpp

```diff
@@ -21,30 +21,73 @@ namespace signalr
         : m_underlying_client(create_client_config(signalr_client_config))
     { }
 
-    pplx::task<void> default_websocket_client::connect(const std::string& url)
+    void default_websocket_client::start(std::string url, transfer_format, std::function<void(std::exception_ptr)> callback)
     {
-        return m_underlying_client.connect(utility::conversions::to_string_t(url));
+        m_underlying_client.connect(utility::conversions::to_string_t(url))
+            .then([callback](pplx::task<void> task)
+            {
+                try
+                {
+                    task.get();
```

Comment:
I think I need a bit more context on what this method is doing. Why do you need to complete the task provided here? Is the callback here only used for exception handling or is it required?

Reply:
This is a point in time thing as we're still using `pplx::task` in our code.

What this is doing, is observing any exceptions from the completed task so we don't crash the process, and then forwarding the exception to the new callback based APIs